Emergence/Contexts/Main Menu/Getting Shows/LocationBasedShowsEmitter.swift

```diff
@@ -0,0 +1,50 @@
+import UIKit
+
+class LocationBasedShowEmitter: NSObject, ShowEmitter {
+    let network: ArtsyProvider<ArtsyAPI>
+    let location: Location
+    let title: String
+    var networking = false
+
+    init(location:Location, network: ArtsyProvider<ArtsyAPI>) {
+        self.network = network
+        self.location = location
+        self.title = location.name
+    }
+
+    var updateBlock: ((shows:[Show]) -> ())?
```

Comment:
Same idea, here. If the update block is something that you want to share, you can give it a more descriptive name and put it in the global scope somewhere, so everyone can access its type.